### PR TITLE
feature: compatible with file.conf and registry.conf configurations

### PR DESCRIPTION
**Pull request by a364176773**
- [ ] I have registered the PR [changes](../changes).

### Ⅱ. Does this pull request fix one issue?

直接复用configurationfactory中load出来的fileconfiguration,fileconfiguration自带对老配置的读取,这怒地fileconfiguration读到的配置转换成springboot可识别的配置进行合并,优先级为自带的application.yml>老配置config,application.yml中没有的配置会从config中读取,全局顺序是-d>env>application.properties/yaml>file.conf&registry.conf
image

### CODE REVIEW

File: server/src/main/java/io/seata/server/spring/listener/SeataPropertiesLoader.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            throw new RuntimeException("load seata registry config error: " + resource.getFilename(), e);
+        }
+    }
+
+    @Override
+    public void initialize(ConfigurableApplicationContext applicationContext) {
+        this.resourceLoader = applicationContext;
+        this.environment = applicationContext.getEnvironment();
+        String registryName = environment.resolvePlaceholders("${SEATA_CONFIG_NAME:}");
+        if (StringUtils.isNotBlank(registryName)) {
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/spring/listener/SeataPropertiesLoader.java:86]: The recommended logic for which conf file to read remains the same:

---

建议对读取哪个conf配置文件的规则与下面保持一致：

https://github.com/seata/seata/blob/8a92c5bb536910206c72ea3d9d257f475195da41/config/seata-config-core/src/main/java/io/s eata/config/ConfigurationFactory.java#L52

Comment (a364176773) [reply server/src/main/java/io/seata/server/spring/listener/SeataPropertiesLoader.java]: ok

File: server/src/main/java/io/seata/server/spring/listener/SeataPropertiesLoader.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        if (CollectionUtils.isNotEmpty(configs)) {
+            Properties properties = new Properties();
+            configs.forEach((k, v) -> {
+                if (v instanceof String) {
+                    if (StringUtils.isEmpty((String)v)) {
+                        return;
+                    }
+                }
+                // Convert the configuration name to the configuration name under Spring Boot
+                if (k.startsWith(FILE_ROOT_PREFIX_REGISTRY) || k.startsWith(FILE_ROOT_PREFIX_CONFIG)) {
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/spring/listener/SeataPropertiesLoader.java:56]: The prefix does not contain the configuration in file.conf.

---

这个前缀没有包含 file.conf的配置。

Comment (slievrly) [reply server/src/main/java/io/seata/server/spring/listener/SeataPropertiesLoader.java:56]: It is recommended to test various scenarios through UT.

Comment (a364176773) [reply server/src/main/java/io/seata/server/spring/listener/SeataPropertiesLoader.java:56]: done

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**